examples/lighting-app/nrfconnect/main/LightingManager.cpp

```diff
@@ -77,6 +77,13 @@ bool LightingManager::InitiateAction(Action_t aAction, int32_t aActor)
         new_state        = kState_Off;
     }
 
+    if (aAction == LEVEL_ACTION)
+    {
+        action_initiated = true;
+        Level(*value); // TODO: check size
```

Comment:
Is there a reason why you don't want to call the LightingManager callbacks for this action? Otherwise, how about setting the level along with On/Off state (line 94)?

Reply:
Also, consider checking size of the binary data in ZclCallbacks.cpp so that LightingManager doesn't care how its state is encoded.

Reply:
> Is there a reason why you don't want to call the LightingManager callbacks for this action?

I think I do: https://github.com/project-chip/connectedhomeip/pull/3806/files#diff-b79e83bc64ae3009aefd8a34dbf93fd4af60a430502455a354235d975b801373R54

I've basically mimic'ed how it is done for On/Off.

> Otherwise, how about setting the level along with On/Off state (line 94)?

Possible, but don't know how well this goes with all the other commands in the LevelControl cluster:
```
#define ZCL_MOVE_TO_LEVEL_COMMAND_ID 0x00             // Ver.: always
#define ZCL_MOVE_COMMAND_ID 0x01                      // Ver.: always
#define ZCL_STEP_COMMAND_ID 0x02                      // Ver.: always
#define ZCL_STOP_COMMAND_ID 0x03                      // Ver.: always
#define ZCL_MOVE_TO_LEVEL_WITH_ON_OFF_COMMAND_ID 0x04 // Ver.: always
#define ZCL_MOVE_WITH_ON_OFF_COMMAND_ID 0x05          // Ver.: always
#define ZCL_STEP_WITH_ON_OFF_COMMAND_ID 0x06          // Ver.: always
#define ZCL_STOP_WITH_ON_OFF_COMMAND_ID 0x07          // Ver.: always
```

Reply:
I meant that you do `return` before calling `mActionInitiated_CB` and `mActionCompleted_CB`. Is that omission intentional?

Reply:
Not intentional. Will fix it.

Reply:
> Is there a reason why you don't want to call the LightingManager callbacks for this action? Otherwise, how about setting the level along with On/Off state (line 94)?

Done.

> Also, consider checking size of the binary data in ZclCallbacks.cpp so that LightingManager doesn't care how its state is encoded.

Done.